core/src/main/java/org/elasticsearch/bootstrap/ESPolicy.java

```diff
@@ -35,19 +36,20 @@
 
 /** custom policy for union of static and dynamic permissions */
 final class ESPolicy extends Policy {
-    
+
     /** template policy file, the one used in tests */
     static final String POLICY_RESOURCE = "security.policy";
     /** limited policy for scripts */
     static final String UNTRUSTED_RESOURCE = "untrusted.policy";
-    
+
     final Policy template;
     final Policy untrusted;
     final Policy system;
     final PermissionCollection dynamic;
     final Map<String,Policy> plugins;
 
     public ESPolicy(PermissionCollection dynamic, Map<String,Policy> plugins, boolean filterBadDefaults) {
+        dynamic.add(new LinkPermission("hard")); // nocommit - is this the right place to add this?
```

Comment:
btw a practical thing might be, add this "merge wrapper" directory with the hard link logic to lucene-misc module. There are other crazy Directories in there already :) Also there is an IndexMergeTool there (https://github.com/apache/lucene-solr/blob/master/lucene/misc/src/java/org/apache/lucene/misc/IndexMergeTool.java) and we could wrap with this hard-link directory by default, so the tool uses less transient space. Maybe removing its forceMerge(1) would also help :)

Just seems to be an ok home for it, to keep things contained.